bin/search-function-test-utilizing-llm.py

```diff
@@ -0,0 +1,105 @@
+import os
+import sys
+import json
+from datetime import date
+
+import ailab.db as db
+from ailab.models import openai
+import ailab.db.finesse as finesse
+
+from ailab.db.finesse.test_queries import get_random_chunk
+
+# Constants
+TEST_VERSION = date.today()
+REQUIRED_QUESTIONS = 1
+CHARACTER_LIMIT = 14383
+STORAGE_PATH = "/home/vscode/finesse-data-2/qna"
```

Comment:
The script should receive these as arguments with default values pointing to these constants. Because the user might want to run the script with different values, especially `storage_path`. Consider using `argparse`.
Example of a script that uses `argparse`:
```python
import argparse

def main(arg1, arg2, arg3):
    # Your main script logic here
    print(f"Argument 1: {arg1}")
    print(f"Argument 2: {arg2}")
    print(f"Argument 3: {arg3}")

def parse_arguments():
    parser = argparse.ArgumentParser(description="Your script description.")
    parser.add_argument("--arg1", default="Default1", help="Description for arg1")
    parser.add_argument("--arg2", default="Default2", help="Description for arg2")
    parser.add_argument("--arg3", default="Default3", help="Description for arg3")
    return parser.parse_args()

if __name__ == "__main__":
    args = parse_arguments()
    main(args.arg1, args.arg2, args.arg3)

```

Running that script would look like:
```bash
python your_script.py --arg1 value1 --arg2 value2 --arg3 value3
```

Make sure to propagate these arguments to the other functions that need them using the default value pattern:
```python
def my_func(arg=DEFAULT_VAL):
    pass
```

Reply:
I'm having trouble with codespaces right now, so I can't test it, but I added the changes.

Reply:
Please let me know when it's complete and tested.

Reply:
It's working